shop3/explicit-stack-search/plan-tree-simple-print.lisp

```diff
@@ -0,0 +1,124 @@
+(in-package :plan-tree)
+
+(defparameter *output-form* nil)
+
+(defmethod print-node ((plan-tree-node top-node))
+  (let ((*output-form* nil))
+    (iter
+     (for node in (complex-tree-node-children plan-tree-node))
+     (format t "~%Node: ~s" node)
+     (print-node node))
+    (format t "~%Plan tree: ~s" *output-form*)))
+  
+(defmethod print-node ((plan-tree-node complex-tree-node))
+  (let ((dependencies (tree-node-dependencies plan-tree-node))
+	output-dependency-form
+	output-children-form)
+    (iter
+     (for d in dependencies)
+     (push
+      `(:establisher
+	((:task
+	  ,(if (eq (establisher d) :init)
+	       :init
+	       (tree-node-task (establisher d))))
+	 (:type ,(type-keyw (type-of (establisher d)))))
+	:label ,(prop d))
+      output-dependency-form))
+
+    (setf children (complex-tree-node-children plan-tree-node))
+    (iter
+     (until (null children))
```

Comment:
`(while children)`

Reply:
Changed in commit e4741f1889b75cd2bc45e191daaceb7b60be935e